cyhy/core/geoloc.py

```diff
@@ -43,11 +43,16 @@ def lookup(self, ip):
 
     def check_restricted_cidr(self, cidr):
         has_restricted = False
-        try:
-            response = self.__reader.city(str(cidr[0]))
-            if response.country.name in RESTRICTED_COUNTRIES:
-                print >> sys.stderr, "Warning! %s traced to restricted country: %s" % (cidr, response.country.name)
-                has_restricted = True
-        except AddressNotFoundError:
-            print >> sys.stderr, "CIDR %s not found in geolocation database" % cidr
+        for ip in cidr:
+            try:
+                response = self.__reader.city(str(ip))
+                if response.country.name in RESTRICTED_COUNTRIES:
+                    has_restricted = True
```

Comment:
@st0rmbl3ss3d - One way to improve this would be to add each restricted IP address to an IPSet, then output the contents of that IPSet at the end.  The IPSet will automatically consolidate the list of addresses into its minimal CIDR form.

For reference:
* https://netaddr.readthedocs.io/en/latest/tutorial_03.html
* https://github.com/cisagov/cyhy-core/search?q=IPSet